dependency/vault_read.go

```diff
@@ -56,70 +58,56 @@ func (d *VaultReadQuery) Fetch(clients *ClientSet, opts *QueryOptions) (interfac
 	case <-d.stopCh:
 		return nil, nil, ErrStopped
 	default:
+		select {
```

Comment:
It took me a few moments to figure out what this select within select was for. I think it could be a little clearer if written like:

```
select{
case <-d.stopCh:
   return nil, nil, ErrStopped
default:
}

select {
case <-d.sleepCh:
default:
}
```

To me its clearer then that what is desired is to return if stopped, but if not stopped to drain the sleepCh. (At least I believe thats the intent here - correct me if I am wrong of course).